conversion/s2s_components/perception.py

```
@@ -0,0 +1,487 @@
+                gw.add_tensor(dst_name, arr.astype(np.float32), raw_dtype=GGMLQuantizationType.F32)
+            elif chosen_qtype == GGMLQuantizationType.F16:
+                gw.add_tensor(dst_name, arr.astype(np.float16), raw_dtype=GGMLQuantizationType.F16)
+            else:
+                packed = _quantize(arr, chosen_qtype)
+                gw.add_tensor(dst_name, packed, raw_dtype=chosen_qtype)
+            dtype_counts[chosen_qtype.name] = dtype_counts.get(chosen_qtype.name, 0) + 1
+            emitted += 1
+
+    print(f"[convert-s2s] emitted {emitted} tensors, skipped {len(skipped)}")
+    print(
+        f"[convert-s2s] dtype tally: "
```

Ruff (0.16.3)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)